specs/lifespan.rst

```diff
@@ -46,6 +46,8 @@ The scope information passed in ``scope`` contains basic metadata:
 * ``asgi["version"]`` (*Unicode string*) -- The version of the ASGI spec.
 * ``asgi["spec_version"]`` (*Unicode string*) -- The version of this spec being
   used. Optional; if missing defaults to ``"1.0"``.
+* ``extensions["state"]`` Optional(*dict[Unicode string, Any]*) -- An empty namespace where
+  the application can persist state to be used when handling subsequent requests.
```

Comment:
Isn't this supposed to be only "state", or did we agree on this "extensions" field previously?

```suggestion
* ``state`` Optional(*dict[Unicode string, Any]*) -- An empty namespace where
  the application can persist state to be used when handling subsequent requests.
```

Reply:
The current spec says that extensions can be used when "a change to a specification is being trialled before being rolled in". That said, I'd be happy to just slap it into the main namespace. It would be a backwards compatible change, i.e. Starlette would just transparently use it instead of it's own dict:
https://github.com/encode/starlette/blob/eb76552d53215de7a319335903dd8a0aebd2df7f/starlette/requests.py#L167-L169. It sounds like your vote would be for main namespace?

Reply:
Oh, no! I'm not in favor of breaking Starlette hahaha

I'm not sure if I get all you said, but what I mean is I don't see the "extensions" namespace being used besides here... :thinking:

Reply:
> I'm not in favor of breaking Starlette hahaha

My point was we would _not_ be breaking Starlette! In fact, using the existing `"state"` key could be a good thing because it means _less_ changes in Starlette to fully support this and some support out of the box as soon as Uvicorn is updated.

> I don't see the "extensions" namespace being used besides here

Do you mean in this section of the docs or in the spec in general? We definitely use it, e.g. `extensions["http.response.trailers"]`.